### PR TITLE
test(app): cover node startup boundaries

### PR DESCRIPTION
**Pull request by hawkingrei**
## Summary

- add focused Rust tests for node-vs-main startup boundaries
- document the verification slice in a dated journal entry
- lock the current startup contract instead of changing runtime behavior

## What changed

- `src/app.rs`
  - add coverage for missing `server.node_id` in node mode
- `src/state.rs`
  - cover root-user creation in main mode
  - cover push/VAPID initialization in main mode
  - keep the existing node-mode skip behavior covered alongside main-mode expectations
- `docs/journal/2026-04-26-node-startup-boundary-verification.md`
  - record scope, validation commands, and follow-up boundary

## Validation

- `cargo test -p agenthub validate_startup_config_ -- --nocapture`
- `cargo test -p agenthub state::tests::setup_database_ -- --nocapture`
- `cargo test -p agenthub state::tests::initialize_services_ -- --nocapture`
- `cargo check -p agenthub`
- `git diff --check`

## Notes

- This slice is intentionally narrow and test-first.
- The current implementation already matched the intended node-startup contract; this PR hardens it with regression coverage and recorded verification.

### CODE REVIEW

File: src/state.rs [hunk 1]
```diff
@@ -671,6 +671,30 @@ mod tests {
...
+        let _guard = ENV_LOCK.lock().await;
+        let temp_home = std::env::temp_dir().join(format!("agenthub-home-{}", Uuid::new_v4()));
+        std::fs::create_dir_all(&temp_home).expect("create temp home");
+        let _home_guard = set_env_var("HOME", &temp_home);
+
+        let config = AppConfig::default();
+
+        let db = AppState::setup_database(&config)
+            .await
+            .expect("setup main database");
```
[the remark this answers is not in the document]
Comment (hawkingrei) [reply src/state.rs:692]: Made the main-role contract explicit in this test by setting `server.role = Main` instead of relying on the default config.

File: src/state.rs [hunk 2]
```diff
@@ -725,6 +749,36 @@ mod tests {
...
+            std::env::temp_dir().join(format!("agenthub-main-startup-{}", Uuid::new_v4()));
+        let keys_path = temp_root.join("push").join("vapid.json");
+        let event_dir = temp_root.join("agent-events");
+        let config = AppConfig {
+            push: Some(PushConfig {
+                subject: Some("mailto:test@example.com".to_string()),
+                keys_path: Some(keys_path.to_string_lossy().to_string()),
+            }),
+            ..Default::default()
+        };
```
[the remark this answers is not in the document]
Comment (hawkingrei) [reply src/state.rs:777]: Updated this test to set `server.role = Main` explicitly so it stays targeted even if defaults change later.
